### ISSUE TITLE
increasing max_history vs. epochs ?

### ISSUE THREAD

**Issue by znat** [RasaHQ/rasa_core#211]
I noticed that as the dialog flow deepens/widens (more branches, more turns), increasing `max_history` or `epochs` both increase accuracy.

I am trying to understand the implication of both.

My understanding, for now, is that increasing `max_history` enriches the model by letting it learn from a longer sequence of dialogue turns (provided examples are long enough), whereas increasing `epochs` improves the memorization of the training and overfits. 

Is that correct?

**Comment by amn41** [comment]
yes that's pretty much right. increasing `max_history` gives your model more capacity to learn (more parameters) and lets it take more history into account.

epochs is how many times we go over the training data in fitting. you can increase this arbitrarily and overfit, but that won't necessarily improve the performance on a test set past a certain point. As you say you are likely to overfit
